tests/checked_set_test.py

```diff
@@ -79,7 +81,27 @@ def test_pickling():
     assert x == y
     assert isinstance(y, Naturals)
 
-
 def test_supports_weakref():
     import weakref
-    weakref.ref(Naturals([1, 2]))
\ No newline at end of file
+    weakref.ref(Naturals([1, 2]))
```

Comment:
It turns out that pyrsistent has had this test and support for weakref structures since @itamarst reported it last year  and it was fixed in in https://github.com/tobgu/pyrsistent/issues/59